executor/ddl.go

```diff
@@ -69,6 +71,39 @@ func (e *DDLExec) toErr(err error) error {
 	return err
 }
 
+// deleteTemporaryTableRecords delete temporary table data.
+func deleteTemporaryTableRecords(memData kv.MemBuffer, tblID int64) error {
```

Comment:
You should also consider dirty data in transaction. Dropping a table should forbids those data to be commited.

May be you can do it in session.doCommit function and filter out data in dropped table